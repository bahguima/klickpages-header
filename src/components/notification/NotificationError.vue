<template>
  <li class="drop__item drop__item--notifications">
    <div class="error-container">
      <span class="error-message">{{errorMessage}}</span>
      <a class="notifications-refresh-btn" href="javascript:" @click="refreshNotifications">
        <i class="fa fa-refresh" />
      </a>
    </div>
  </li>
</template>

<script>
export default {
  props: {
    errorMessage: String,
  },
  methods: {
    refreshNotifications(event) {
      event.stopPropagation();
      this.$emit('refreshNotifications');
    },
  },
};
</script>
