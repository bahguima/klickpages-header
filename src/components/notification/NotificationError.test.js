import { createLocalVue, shallowMount } from '@vue/test-utils';

import NotificationError from './NotificationError.vue';

describe('components/notification/NotificationError', () => {
  const localVue = createLocalVue();
  let wrapper;

  describe('when components is mounted', () => {
    const propsData = {
      errorMessage: 'test error message',
    };

    beforeAll(() => {
      wrapper = shallowMount(NotificationError, { localVue, propsData });
    });

    afterAll(() => {
      wrapper.destroy();
    });

    it('should renders html correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('refreshNotifications', () => {
      const event = {
        stopPropagation: jest.fn(),
      };

      beforeAll(() => {
        wrapper = shallowMount(NotificationError, { localVue });
        wrapper.vm.refreshNotifications(event);
      });

      afterAll(() => {
        wrapper.destroy();
      });

      it('should stop event propagation', () => {
        expect(event.stopPropagation).toHaveBeenCalled();
      });

      it('should emmit refreshNotifications event', () => {
        expect(wrapper.emitted().refreshNotifications).toBeTruthy();
      });
    });
  });
});
