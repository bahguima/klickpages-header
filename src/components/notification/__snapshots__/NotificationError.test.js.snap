// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/notification/NotificationError when components is mounted should renders html correctly 1`] = `
<li
  class="drop__item drop__item--notifications"
>
  <div
    class="error-container"
  >
    <span
      class="error-message"
    >
      test error message
    </span>
     
    <a
      class="notifications-refresh-btn"
      href="javascript:"
    >
      <i
        class="fa fa-refresh"
      />
    </a>
  </div>
</li>
`;
